Zend/zend_generators.c

```diff
@@ -330,6 +325,10 @@ static void zend_generator_free_storage(zend_object *object) /* {{{ */
 
 	zend_generator_close(generator, 0);
 
+	if (generator->func && (generator->func->common.fn_flags & ZEND_ACC_CLOSURE)) {
```

Comment:
Is the `generator->func` check needed? When can it be `NULL`?

Reply:
Yes, it is required: https://github.com/php/php-src/pull/14167#issuecomment-2107035386.

`generator->func == NULL` when attempting to manually instantiate a Generator object via `new Generator()`.

Reply:
Ugh, why is that allowed... But ok!

Reply:
It’s not allowed: [It throws](https://github.com/php/php-src/blob/master/Zend/tests/generators/errors/generator_instantiate_error.phpt). But at the point where it throws the object is already initialized so far that `zend_generator_free_storage()` is called.

Reply:
Ah, ok...